File: sample/src/main/java/com/afollestad/aestheticsample/MainAdapter.java
package com.afollestad.aestheticsample;

import android.annotation.SuppressLint;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;

/** @author Aidan Follestad (afollestad) */
class MainAdapter extends RecyclerView.Adapter<MainAdapter.ViewHolder> {

  @NonNull
  @Override
  public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    View view =
        LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_rv, parent, false);
    return new ViewHolder(view);
  }

  @SuppressLint("SetTextI18n")
  @Override
  public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
    holder.title.setText("Item #" + position);
    holder.subtitle.setText(R.string.hello_world);
  }

  @Override
  public int getItemCount() {
    return 20;
  }

  static class ViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.title)
    TextView title;

    @BindView(R.id.subtitle)
    TextView subtitle;

    ViewHolder(View itemView) {
      super(itemView);
      ButterKnife.bind(this, itemView);
    }
  }
}

File: sample/src/main/java/com/afollestad/aestheticsample/MainAdapter.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.aestheticsample

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.list_item_rv.view.subtitle
import kotlinx.android.synthetic.main.list_item_rv.view.title

/** @author Aidan Follestad (afollestad) */
internal class MainAdapter : RecyclerView.Adapter<MainAdapter.ViewHolder>() {

  override fun onCreateViewHolder(
    parent: ViewGroup,
    viewType: Int
  ): ViewHolder {
    val view = LayoutInflater.from(parent.context)
        .inflate(R.layout.list_item_rv, parent, false)
    return ViewHolder(view)
  }

  @SuppressLint("SetTextI18n")
  override fun onBindViewHolder(
    holder: ViewHolder,
    position: Int
  ) {
    holder.itemView.title.text = "Item #$position"
    holder.itemView.subtitle.setText(R.string.hello_world)
  }

  override fun getItemCount(): Int {
    return 20
  }

  internal class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}
